src/AssetTransferApi.ts

```diff
@@ -294,8 +307,11 @@ export class AssetTransferApi {
 		let txMethod: Methods;
 		let transaction: SubmittableExtrinsic<'promise', ISubmittableResult>;
 
-		if (xcmPallet === XcmPalletName.xTokens && xcmDirection === Direction.ParaToSystem) {
-			if (paysWithFeeDest && !paysWithFeeDest.includes('parents') && assetIds.length < 2) {
+		if (
+			xcmPallet === XcmPalletName.xTokens &&
+			(xcmDirection === Direction.ParaToSystem || xcmDirection === Direction.ParaToPara)
+		) {
+			if (!paysWithFeeDest && assetIds.length < 2) {
```

Comment:
Can we do assetIds.length < 2 in one of our layer checks? Maybe in `checkBaseInputTypes`?

Reply:
Great question! After looking at it a bit I don't think it would work as an alternative place to check the assetIds length since in this case we aren't checking the length for validity but more as a way to determine which `xTokens` method to use. Definitely open to discuss it more though

Reply:
Ahh great point, that totally makes sense. Lets keep the check as it is then! Thanks for the explanation.